src/utils_helpers.js

```diff
@@ -82,3 +101,37 @@ function zeros(length){
   return Array.from(empty_mat, arr => arr.fill(0))
 }
 
+// helpers to wrangle data for the balance plot
+function balanceDat(elem1, elem2) {
+  const types_1 = elem1.map(d => d.types)
+  const types_2 = elem2.map(d => d.types)
+  
+  const union_types = getUnions(types_1, types_2)
+  const tot_types = types_1.length+types_2.length
+  
+  return [ 
+    { y_coord: "total count",     frequency: +(types_2.length / tot_types).toFixed(3) },
+    { y_coord: "total count",     frequency: -(types_1.length / tot_types).toFixed(3) },
+    { y_coord: "all names",       frequency: +(types_2.length / union_types.size).toFixed(3) },
+    { y_coord: "all names",       frequency: -(types_1.length / union_types.size).toFixed(3) },
+    { y_coord: "exclusive names", frequency: +(setdiff(types_2, types_1).size / types_2.length).toFixed(3) },
+    { y_coord: "exclusive names", frequency: -(setdiff(types_1, types_2).size / types_1.length).toFixed(3) } 
+  ]
+}
+   
+// helper to wrangle the data for the wordshift plot
+function wordShift_dat(me, dat) { 
+  const out = []
+  for (let i=0; i < me[0]['types'].length; i++) {
+    const rank_diff = me[0]['ranks'][i]-me[1]['ranks'][i]
+    out.push({
+      'type': `${me[0]['types'][i]} (${me[0]['ranks'][i]} ⇋ ${me[1]['ranks'][i]})` ,
+      'rank_diff': rank_diff,
+      'metric': rank_diff < 0 ? -dat.deltas[i] : dat.deltas[i], 
+    })
+  }
+  
+  return out.slice().sort((a, b) => descending(Math.abs(a.metric), Math.abs(b.metric)))
+}
```

Comment:
These helpers don't feel general enough to go in `utils`. I suggest putting them in `combine_distributions.js` because they're operating specifically on mixed elements calculations.

Reply:
done